arbnode/message-extraction/extractor_loop.go

```diff
@@ -124,6 +181,18 @@ func (m *MessageExtractor) Act(ctx context.Context) error {
 		}
 		preState := processAction.melState
 
+		// Check for reorg
+		wasReorged, err := m.wasBlockReorgedOut(ctx, preState.ParentChainBlockNumber, preState.ParentChainBlockHash)
+		if err != nil {
+			return err
+		}
+		if wasReorged {
+			// Found a reorg, we move to ReorgingToOldBlock state to handle it
+			return m.fsm.Do(reorgingToOldBlock{
+				melState: preState,
```

Comment:
Remove this logic, and instead reorg after line 198 and pass in the parentChainBlock according to the comment above